saleor/dashboard/product/forms.py

```diff
@@ -241,7 +241,8 @@ class Meta:
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         product_type = self.instance.product_type
-        self.initial['tax_rate'] = product_type.tax_rate
+        if self.instance.tax_rate:
```

Comment:
If `instance.tax_rate` is None we want to render the default from the product type. I'd suggest:

```
self.initial['tax_rate'] = self.instance.tax_rate or product_type.tax_rate
```